install.sh

```diff
@@ -94,133 +103,128 @@ esac
 
 # functions to maintain packages
 function download_check () {
-    cd $CREW_BREW_DIR
+    cd "${CREW_BREW_DIR}"
 
     #download
-    echo "Downloading $1..."
-    curl --progress-bar -C - -# -L --ssl $2 -o "$3"
+    echo "Downloading ${1}..."
+    curl --progress-bar -C - -L --ssl "${2}" -o "${3}"
 
     #verify
-    echo "Verifying $1..."
-    echo $4 $3 | sha256sum -c -
-    case $? in
+    echo "Verifying ${1}..."
+    echo "${4}" "${3}" | sha256sum -c -
+    case "${?}" in
     0) ;;
     *)
-      echo "Verification failed, something may be wrong with the $1 download."
+      echo "Verification failed, something may be wrong with the download."
       exit 1;;
     esac
 }
 
 function extract_install () {
-    cd $CREW_BREW_DIR
+    cd "${CREW_BREW_DIR}/dest"
 
     #extract and install
-    echo "Extracting $1 (this may take a while)..."
-    rm -rf ./usr ./home
-    tar -xf $2
-    echo "Installing $1 (this may take a while)..."
-    if [ -d "${CREW_BREW_DIR}/home" ]; then
-      tar cf - ./usr/* ./home/* | (cd /; tar xp --keep-directory-symlink -f -)
-    else
-      tar cf - ./usr/* | (cd /; tar xp --keep-directory-symlink -f -)
-    fi
-    mv ./dlist $CREW_CONFIG_PATH/meta/$1.directorylist
-    mv ./filelist $CREW_CONFIG_PATH/meta/$1.filelist
+    echo "Extracting ${1} (this may take a while)..."
+    for dir in ./*/; do
+      rm -rf "${dir}"
+    done
+    for file in ./*list; do
+      rm -f "${file}"
+    done
+    tar xpf ../"${2}"
+    echo "Installing ${1} (this may take a while)..."
+    tar cpf - ./*/* | (cd /; tar xp --keep-directory-symlink -f -)
+    mv ./dlist "${CREW_CONFIG_PATH}/meta/${1}.directorylist"
+    mv ./filelist "${CREW_CONFIG_PATH}/meta/${1}.filelist"
```

Comment:
I don't understand why this section was changed.

Reply:
Because there might be other directories than just `/usr` and `/home` that have files in them in the future, and because we want to preserve permissions for those.